apps/web/src/pages/integrations/components/ChannelGroup.tsx

```diff
@@ -1,17 +1,30 @@
 import React from 'react';
 import { Grid } from '@mantine/core';
+import { IProviderConfig } from '@notifire/shared';
 import { ProviderCard } from './ProviderCard';
 import { Title } from '../../../design-system';
 
-export function ChannelGroup({ title, providers }: { providers: any[]; title: string }) {
+export function ChannelGroup({
+  title,
+  providers,
+  showModalData,
+}: {
+  providers: any[];
+  title: string;
+  showModalData: (visible: boolean, provider: IProviderConfig) => void;
```

Comment:
I would go with something like onProviderClick to make it more generic in case a different than modal interaction will be used down the road. 
```suggestion
  onProviderClick: (visible: boolean, provider: IProviderConfig) => void;
```

Reply:
Good idea I will update it! :)